src/oca/python/pyone/acl.py

```diff
@@ -0,0 +1,195 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Abstract rules of the type USER RESOURCE RIGHTS
+# which are:
+#     USER      -> #<num>
+#                  @<num>
+#                  ALL
+#     RESOURCE  -> + separated list and "/{#,@,%}<num>|ALL"
+#                  VM
+#                  HOST
+#                  NET
+#                  IMAGE
+#                  USER
+#                  TEMPLATE
+#                  GROUP
+#                  DATASTORE
+#                  CLUSTER
+#                  DOCUMENT
+#                  ZONE
+#                  SECGROUP
+#                  VDC
+#                  VROUTER
+#                  MARKETPLACE
+#                  MARKETPLACEAPP
+#                  VMGROUP
+#     RIGHTS    -> + separated list
+#                  USE
+#                  MANAGE
+#                  ADMIN
+#                  CREATE
+import re
+
+USERS = {
+    "UID": 0x100000000,
+    "GID": 0x200000000,
+    "ALL": 0x400000000,
+    "CLUSTER": 0x800000000
+}
+
+RESOURCES = {
+    "VM" :     0x1000000000,
+    "HOST":     0x2000000000,
+    "NET":     0x4000000000,
+    "IMAGE":     0x8000000000,
+    "USER":    0x10000000000,
+    "TEMPLATE":    0x20000000000,
+    "GROUP":    0x40000000000,
+    "DATASTORE":   0x100000000000,
+    "CLUSTER":   0x200000000000,
+    "DOCUMENT":   0x400000000000,
+    "ZONE":   0x800000000000,
+    "SECGROUP":  0x1000000000000,
+    "VDC":  0x2000000000000,
+    "VROUTER":  0x4000000000000,
+    "MARKETPLACE":  0x8000000000000,
+    "MARKETPLACEAPP": 0x10000000000000,
+    "VMGROUP": 0x20000000000000,
+    "VNTEMPLATE": 0x40000000000000
+}
+
+RIGHTS = {
+    "USE": 0x1,  # Auth. to use an object
+    "MANAGE": 0x2,  # Auth. to perform management actions
+    "ADMIN": 0x4,  # Auth. to perform administrative actions
+    "CREATE": 0x8   # Auth. to create an object
+}
+
+
+class OneAcl():
+    # Converts a string in the form [#<id>, @<id>, *] to a hex. number
+    #
+    # @param users [String] Users component string
+    #
+    # @return [String] A string containing a hex number
+    def parse_users(self, users):
+       return hex(self.calculate_ids(users))
+
+
+    # Converts a resources string to a hex. number
+    #
+    # @param resources [String] Resources component string
+    #
+    # @return [String] A string containing a hex number
+    def parse_resources(self, resources):
+        ret = 0
+        resources = resources.split("/")
+
+        if len(resources) != 2:
+            raise Exception("Resource '#{resources}' malformed")
```

Comment:
this is leftover from ruby? can you change to be python2/3 compatible?
(also below)